packages/table-core/src/features/RowSelection.ts

```diff
@@ -353,7 +353,8 @@ export const RowSelection: TableFeature = {
         table.setRowSelection(old => {
           value = typeof value !== 'undefined' ? value : !isSelected
 
-          if (isSelected === value) {
+          const willUpdateSubRows = row.subRows?.length && row.getCanSelectSubRows()
```

Comment:
**row.subRows?.length** can be undefined, how about using !! to explicitly convert it to a boolean?